brainscore_vision/model_helpers/activations/core.py

```diff
@@ -125,31 +177,99 @@ def register_stimulus_set_hook(self, hook):
             batch_end = min(batch_start + batch_size, len(paths))
             batch_inputs = paths[batch_start:batch_end]
-            batch_activations = self._get_batch_activations(batch_inputs, layer_names=layers, batch_size=batch_size)
-            for hook in self._batch_activations_hooks.copy().values():  # copy to avoid handle re-enabling messing with the loop
+
+            batch_activations = OrderedDict()
+            # compute activations on the entire batch one shift at a time
+            for shift_number in range(self.number_of_trials):
+
+                activations = self._get_batch_activations(batch_inputs, layers, batch_size, require_variance,
+                                                          shift_number)
+
+                for layer_name, layer_output in activations.items():
+                    batch_activations.setdefault(layer_name, []).append(layer_output)
+
+            # concatenate all shifts into this batch
+            for layer_name, layer_outputs in batch_activations.items():
+                batch_activations[layer_name] = np.concatenate(layer_outputs)
+
+            for hook in self._batch_activations_hooks.copy().values():
                 batch_activations = hook(batch_activations)
 
-            if layer_activations is None:
-                layer_activations = copy.copy(batch_activations)
-            else:
-                for layer_name, layer_output in batch_activations.items():
-                    layer_activations[layer_name] = np.concatenate((layer_activations[layer_name], layer_output))
+            # add this batch to layer_activations
+            for layer_name, layer_output in batch_activations.items():
+                layer_activations.setdefault(layer_name, []).append(layer_output)
 
-        return layer_activations
+        # fast concat all batches
+        for layer_name, layer_outputs in layer_activations.items():
+            layer_activations[layer_name] = np.concatenate(layer_outputs)
 
-    def _get_batch_activations(self, inputs, layer_names, batch_size):
+        return layer_activations  # this is all batches
+
+    def _get_batch_activations(self, inputs, layer_names, batch_size, require_variance: bool = False,
+                               trial_number: int = 1):
         inputs, num_padding = self._pad(inputs, batch_size)
         preprocessed_inputs = self.preprocess(inputs)
+        preprocessed_inputs = self.translate_images(preprocessed_inputs, inputs, trial_number, require_variance)
         activations = self.get_activations(preprocessed_inputs, layer_names)
         assert isinstance(activations, OrderedDict)
         activations = self._unpad(activations, num_padding)
+        if require_variance:
+            self.remove_temporary_files(preprocessed_inputs)
         return activations
 
+    def set_visual_degrees(self, visual_degrees: float) -> None:
```

Comment:
```suggestion
    def set_visual_degrees(self, visual_degrees: float):
```
usually we wouldn't write the `None` for a function not returning anything